compiler/fir/tree/src/org/jetbrains/kotlin/fir/declarations/FirDeclarationUtil.kt

```diff
@@ -116,6 +116,8 @@ fun FirRegularClass.addDeclaration(declaration: FirDeclaration) {
 
 private object IsFromVarargKey : FirDeclarationDataKey()
 var FirProperty.isFromVararg: Boolean? by FirDeclarationDataRegistry.data(IsFromVarargKey)
+private object IsReferredViaField: FirDeclarationDataKey()
+var FirProperty.isReferredViaField: Boolean? by FirDeclarationDataRegistry.data(IsReferredViaField)
```

Comment:
Well, it may be possible but it's better to avoid such code....

Reply:
Not sure you don't like this kind of attribute addition or marking the presence of backing field for a property. As per the TODO in the conversion: `FIR resolve should determine backing field presence itself`, it is okay to mark the presence of backing field somehow...? E.g., adding a bit to `FirProperty` or encoding as a resolved state change of `FirProperty`?

Reply:
Tomorrow we will discuss it and decide something.

Reply:
After seeing how resolve state works, I found it doesn't make sense to add the presence of a backing field as a resolved state. Also, instead of `FirProperty`, I found that a backing field symbol is always bound, so when that symbol is used to create a backing field reference, actually that is a good timing to indicate that this property's backing field is referenced. Hence, added a flag to the symbol.